src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -502,27 +502,33 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
         }
     }, [backend, sendAlert]);
 
-    const kill = useCallback(async () => {
+    const kill = useCallback(() => {
         try {
             setStatus(ExecutionStatus.KILLING);
-            const backendKillPromise = backend.kill();
-            const timeoutPromise = delay(2500).then(() => ({
-                type: 'timeout',
-                exception: '',
-            }));
-            const response = await Promise.race([backendKillPromise, timeoutPromise]);
-            if (response.type === 'timeout') {
-                await restart();
-                log.info('Finished restarting backend');
-            }
-            if (response.type === 'error') {
-                sendAlert({ type: AlertType.ERROR, message: response.exception });
-            }
+            backend.abort();
+            // Give it a second to abort the run request before submitting another request
+            // TODO: the abort and the timeout can be removed once `run` is made fully async
+            setTimeout(() => {
```

Comment:
I explained it in the comment. Remove this settimeout and try for yourself. It won't actually kill the backend

Reply:
Well, then that means that killing itself is bugged, because this timeout should not be necessary.

Reply:
I'm not 100% confident, but I'm pretty sure it's a race condition from aborting the run request. Like, it triggers the abort, then tries to run the kill request immediately, but the abortcontroller hasn't been reset yet, so it just auto aborts that request before it even starts. So, putting in the timeout gives it time to reset the abort controller. 

If you have a better idea of what's going on, please give me an actual suggestion on what to do instead, because right now I don't see another way to fix this aside from changing how run works to not require an abort at all. (The reason it needs an abort is because without it, the request fails and causes an error to pop up in the UI)

Reply:
> Like, it triggers the abort, then tries to run the kill request immediately, but the abortcontroller hasn't been reset yet

`abort` swaps out the old abort controller, so this can't happen. https://github.com/chaiNNer-org/chaiNNer/blob/4772713ea9e59f3950ef0c48d3d818c6c50c3ae1/src/common/Backend.ts#L211